Wrappers/Python/ccpi/optimisation/functions/FunctionOperatorComposition.py

```diff
@@ -50,25 +50,26 @@ def __init__(self, function, operator):
     :type f: :code:`Function`
     '''
 
-        super(FunctionOperatorComposition, self).__init__(L=None)
+        super(FunctionOperatorComposition, self).__init__()
         
-        self.function = function     
-        self.operator = operator
-        
-        if not isinstance(self.function, Function):
+        if not isinstance(function, Function):
             raise ValueError('{} is not function '.format(type(self.function)))
-          
-        # check also ScaledOperator because it's not a child atm of Operator            
-        if not isinstance(self.operator, (Operator, ScaledOperator)):
+
+        if not isinstance(operator, Operator):
             raise ValueError('{} is not function '.format(type(self.operator)))            
         
-        if self.L is None:
+        self.function = function     
+        self.operator = operator
+
+    @property
+    def L(self):
+        if self._L is None:
             try:
-                self.L = function.L * operator.norm()**2 
+                return self.function.L * self.operator.norm()**2 
```

Comment:
This line calls the expensive `norm` method of the operator. Is this only executed when the user asks for `f.L` explicitly, and not during initialisation of the function `f`?

Reply:
I suppose this should set `self._L` and then return it

```python
@property
def L(self):
    if self._L is None:
        try:
            self._L = self.function.L  * (self.operator.norm() ** 2)
        except ValueError as ve:
            self._L = None
    return self._L
    
```